pkg/settings/setting.go

```diff
@@ -236,6 +239,130 @@ var (
 	UIPreferred = NewSetting("ui-preferred", "vue")
 )
 
+func getClusterAgentDefaultAffinity() string {
+	// set default to affinity that cluster agent currently uses
+	defaultAffinity := v1.Affinity{
+		PodAntiAffinity: &v1.PodAntiAffinity{
+			PreferredDuringSchedulingIgnoredDuringExecution: []v1.WeightedPodAffinityTerm{
+				{
+					Weight: 100,
+					PodAffinityTerm: v1.PodAffinityTerm{
+						LabelSelector: &metav1.LabelSelector{
+							MatchExpressions: []metav1.LabelSelectorRequirement{
+								{
+									Key:      "app",
+									Operator: metav1.LabelSelectorOperator(v1.NodeSelectorOpIn),
+									Values:   []string{"cattle-cluster-agent"},
+								},
+							},
+						},
+						TopologyKey: "kubernetes.io/hostname",
+					},
+				},
+			},
+		},
+		NodeAffinity: &v1.NodeAffinity{
+			RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
+				NodeSelectorTerms: []v1.NodeSelectorTerm{
+					{
+						MatchExpressions: []v1.NodeSelectorRequirement{
+							{
+								Key:      "beta.kubernetes.io/os",
+								Operator: v1.NodeSelectorOpNotIn,
+								Values:   []string{"windows"},
+							},
+						},
+					},
+				},
+			},
+			PreferredDuringSchedulingIgnoredDuringExecution: []v1.PreferredSchedulingTerm{
+				{
+					Weight: 100,
+					Preference: v1.NodeSelectorTerm{
+						MatchExpressions: []v1.NodeSelectorRequirement{
+							{
+								Key:      "node-role.kubernetes.io/controlplane",
+								Operator: v1.NodeSelectorOpIn,
+								Values:   []string{"true"},
+							},
+						},
+					},
+				},
+				{
+					Weight: 100,
+					Preference: v1.NodeSelectorTerm{
+						MatchExpressions: []v1.NodeSelectorRequirement{
+							{
+								Key:      "node-role.kubernetes.io/control-plane",
+								Operator: v1.NodeSelectorOpIn,
+								Values:   []string{"true"},
+							},
+						},
+					},
+				},
+				{
+					Weight: 100,
+					Preference: v1.NodeSelectorTerm{
+						MatchExpressions: []v1.NodeSelectorRequirement{
+							{
+								Key:      "node-role.kubernetes.io/master",
+								Operator: v1.NodeSelectorOpIn,
+								Values:   []string{"true"},
+							},
+						},
+					},
+				},
+				{
+					Weight: 1,
+					Preference: v1.NodeSelectorTerm{
+						MatchExpressions: []v1.NodeSelectorRequirement{
+							{
+								Key:      "cattle.io/cluster-agent",
+								Operator: v1.NodeSelectorOpIn,
+								Values:   []string{"true"},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	data, err := json.Marshal(defaultAffinity)
+	if err != nil {
+		logrus.Errorf("failed to marshal cluster agent affinity: %s", err)
+	}
+	return string(data)
+}
+
+func getFleetAgentDefaultAffinity() string {
```

Comment:
I worry that these are going to fall out of date -- what kind of process are we going to put in place to keep the defaults we set here updated with what Fleet defines as their default values?

Reply:
May more be a question for the Fleet team

Reply:
That is part of the current design and I pinged Fleet devs to check on the process to keep these updated.
Luckily default affinity didn't change in the last 2 years or so (now that I said it it's probably going to change next month).